inference-engine/src/mkldnn_plugin/nodes/mkldnn_topk_node.cpp

```diff
@@ -42,128 +1197,113 @@ bool MKLDNNTopKNode::isSupportedOperation(const std::shared_ptr<const ngraph::No
     return true;
 }
 
-MKLDNNTopKNode::MKLDNNTopKNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng,
-                                     MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+MKLDNNTopKNode::MKLDNNTopKNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(op, eng, cache) {
     std::string errorMessage;
-    if (!isSupportedOperation(op, errorMessage)) {
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "TopK layer with name '" + getName() + "'";
+
+        auto topKOp = ngraph::as_type_ptr<ngraph::op::v1::TopK>(op);
+
+        in_dims = topKOp->get_input_partial_shape(TOPK_DATA);
+        out_dims = topKOp->get_output_partial_shape(TOPK_DATA);
+        out_idx_dims = topKOp->get_output_partial_shape(TOPK_INDEX);
+        in_dims_size = in_dims.size();
+        out_dims_size = out_dims.size();
+
+        if (!isDynamicNgraphNode(op)) {
+            auto topKConst = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(topKOp->get_input_node_shared_ptr(TOPK_K));
+            if (!topKConst) {
+                IE_THROW() << errorPrefix <<  "gets non-constant second tensor in static shape mode!";
+            }
+            top_k = topKConst->cast_vector<int>()[0];
+        }
+
+        axis = topKOp->get_axis();
+        mode_max = topKOp->get_mode() == ngraph::op::TopKMode::MAX;
+        sort_index = topKOp->get_sort_type() == ngraph::op::TopKSortType::SORT_INDICES;
+    } else {
         IE_THROW(NotImplemented) << errorMessage;
     }
-    auto topK1Op = ngraph::as_type_ptr<ngraph::op::v1::TopK>(op);
-
-    axis = topK1Op->get_axis();
-    mode_max = topK1Op->get_mode() == ngraph::op::TopKMode::MAX;
-    sort_value = topK1Op->get_sort_type() == ngraph::op::TopKSortType::SORT_VALUES;
 }
 
-void MKLDNNTopKNode::initSupportedPrimitiveDescriptors() {
-    if (!supportedPrimitiveDescriptors.empty())
+void MKLDNNTopKNode::getSupportedDescriptors() {
```

Comment:
Could you please move all checks inside getSupportedDescriptors() to ctor? getSupportedDescriptors just designed for other purposes and expected to be used only for oneDNN specific nodes.

Reply:
Fixed. Thanks Dmitry!